optuna/storages/in_memory.py

```diff
@@ -169,16 +169,22 @@ def _create_running_trial():
             datetime_complete=None)
 
     def set_trial_state(self, trial_id, state):
-        # type: (int, structs.TrialState) -> None
+        # type: (int, structs.TrialState) -> bool
 
         with self._lock:
             self.check_trial_is_updatable(trial_id, self.trials[trial_id].state)
```

Comment:
Currently this raises RuntimeError when `trial.is_finished`. So it needs to update the condition.